### ISSUE TITLE
Validation is run after every batch when `trainer.should_stop==True` but `min_epochs` is not met

### ISSUE THREAD

**Issue by leokraft** [Lightning-AI/lightning#16363]
### Bug description

This bug is encountered when using the `EarlyStopping` callback and setting `min_epochs` in the `Trainer`.

When early stopping is triggered `should_stop=True` is set in the `Trainer` ([here](https://github.com/Lightning-AI/lightning/blob/3828f527dd348d414dcaffe8d07aea1f689cfb22/src/pytorch_lightning/callbacks/early_stopping.py#L205)).
However if `min_epochs` is not reached [`_should_stop_early`](https://github.com/Lightning-AI/lightning/blob/3828f527dd348d414dcaffe8d07aea1f689cfb22/src/pytorch_lightning/loops/fit_loop.py#L150) is`False` and we get the following, expected, message ([origin](https://github.com/Lightning-AI/lightning/blob/3828f527dd348d414dcaffe8d07aea1f689cfb22/src/pytorch_lightning/loops/epoch/training_epoch_loop.py#L108)):

```sh
INFO:pytorch_lightning.utilities.rank_zero:Trainer was signaled to stop but the required `min_epochs=...` or `min_steps=...` has not been met. Training will continue...
```

Afterwards the training will continue since `_should_stop_early==False`.

Since `should_stop` is now set but the training continues, a validation run will be triggered after each batch in the `_should_check_val_fx` method. (see [here](https://github.com/Lightning-AI/lightning/blob/3828f527dd348d414dcaffe8d07aea1f689cfb22/src/pytorch_lightning/loops/epoch/training_epoch_loop.py#L485))

This happens because of

```py
def _should_check_val_fx(self) -> bool:
    ...
    if self.trainer.should_stop:
            return True
    ...
```

One solution I could think of would be

```py
def _should_check_val_fx(self) -> bool:
    ...
    if self.trainer.should_stop and self.trainer.fit_loop._should_stop_early:
            return True
    ...
```

However I am not deep enough into lightning to figure if this breaks anything else.
I am looking forward to your comments and suggestions.

### How to reproduce the bug
A working example can be found in this [Colab](https://colab.research.google.com/gist/leokraft/76da9bc8819079dfbaf6d3a3cf812657/bug_report_model.ipynb).

### Error messages and logs

```
INFO:pytorch_lightning.utilities.rank_zero:Trainer was signaled to stop but the required `min_epochs=...` or `min_steps=...` has not been met. Training will continue...
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by coryan-umd** [comment]
Thanks @leokraft for the suggested solution, I have also encountered this issue.

**Comment by romankouz** [comment]
I can confirm that this error still exists. This has the unintended consequence of ignoring the `log_ever_n_steps` flag and records the validation loss every step instead.
